p/scripts/main.js

```diff
@@ -720,11 +720,24 @@ function onScroll() {
 			}
 		});
 	}
-	if (context.auto_load_more) {
-		const streamFooter = document.getElementById('stream-footer');
-		if (streamFooter && box_to_follow.offsetHeight > 0 &&
+	let streamFooter;
+	if (context.auto_load_more && (streamFooter = document.getElementById('stream-footer'))) {
+		if (box_to_follow.offsetHeight > 0 &&
 			box_to_follow.scrollTop + box_to_follow.offsetHeight + (window.innerHeight / 2) >= streamFooter.offsetTop) {
+			// Too close to the last pre-loaded article
 			load_more_posts();
+		} else {
+			let sibling = document.querySelector('.flux.current');
+			if (sibling) {
+				for (let i = 5; i > 0; i--) {
+					sibling = sibling.nextElementSibling;
+					if (!sibling) {
+						// Too few pre-loaded articles
+						load_more_posts();
+						break;
+					}
+				}
+			}
 		}
```

Comment:
I like what you did here.
I think you can simplify that with:
```javascript
} elseif (document.querySelectorAll('.flux.current ~ .flux').length <= 5) {
    load_more_posts();
}
```

I cannot find a feed to reproduce on my dev instance. I'll get back to you when I find one.

Reply:
I could reproduce the bug with the current branch using this feed (https://hackaday.com/blog/feed/). Your solution fixes the bug. Mine too.

On another topic, I think we should add some marker on the last article to alert the user that it will switch feed after the current article.

Reply:
FYI https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator